Products/PasswordStrength/plugin.py

```diff
@@ -1,280 +1,242 @@
+    security = ClassSecurityInfo()
+
+    _properties = ({'id': 'title',
+                    'label': 'Title',
+                    'type': 'string',
+                    'mode': 'w',
+                    },
+                   {'id': 'p1_re',
+                    'label': 'Policy 1 Regular Expression',
+                    'type': 'string',
+                    'mode': 'w',
+                    },
+                   {'id': 'p1_err',
+                    'label': 'Policy 1 Error Message',
+                    'type': 'string',
+                    'mode': 'w',
+                    },
+                   {'id': 'p2_re',
+                    'label': 'Policy 2 Regular Expression',
+                    'type': 'string',
+                    'mode': 'w',
+                    },
+                   {'id': 'p2_err',
+                    'label': 'Policy 2 Error Message',
+                    'type': 'string',
+                    'mode': 'w',
+                    },
+                   {'id': 'p3_re',
+                    'label': 'Policy 3 Regular Expression',
+                    'type': 'string',
+                    'mode': 'w',
+                    },
+                   {'id': 'p3_err',
+                    'label': 'Policy 3 Error Message',
+                    'type': 'string',
+                    'mode': 'w',
+                    },
+                   {'id': 'p4_re',
+                    'label': 'Policy 4 Regular Expression',
+                    'type': 'string',
+                    'mode': 'w',
+                    },
+                   {'id': 'p4_err',
+                    'label': 'Policy 4 Error Message',
+                    'type': 'string',
+                    'mode': 'w',
+                    },
+                   {'id': 'p5_re',
+                    'label': 'Policy 5 Regular Expression',
+                    'type': 'string',
+                    'mode': 'w',
+                    },
+                   {'id': 'p5_err',
+                    'label': 'Policy 5 Error Message',
+                    'type': 'string',
+                    'mode': 'w',
+                    },
+                   )
+
+    def __init__(self, id, title=None):
+        self._id = self.id = id
+        self.title = title
+
+        i = 1
+        for reg, err in DEFAULT_POLICIES:
+            setattr(self, 'p%i_re' % i, reg)
+            setattr(self, 'p%i_err' % i, err)
+            i += 1
+
+    security.declarePrivate('validateUserInfo')
+    def validateUserInfo(self, user, set_id, set_info):
+        """ -> ( error_info_1, ... error_info_N )
+
+        o Returned values are dictionaries, containing at least keys:
+
+          'id' -- the ID of the property, or None if the error is not
+                  specific to one property.
+
+          'error' -- the message string, suitable for display to the user.
+        """
+
+        errors = []
+        site = portal.getSite()
+        if set_info and set_info.get('password', None) is not None:
```

Comment:
Another instance where you can reduce cyclomatic complexity by enforcing a valid context:

```python
if not set_info and not set_info.get('password', None):
    return errors

#...your other stuff...
```